packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorViewImpl.java

```diff
@@ -558,9 +558,15 @@ private ExpressionProps generateExpressionProps(final ExpressionEditorDefinition
      * @param expressionCommand
      */
     void executeUndoableExpressionCommand(final FillExpressionCommand expressionCommand) {
+
         createUndoCommand();
+        if (!expressionCommand.isCurrentExpressionOfTheSameType()) {
```

Comment:
We simulate the action of "clear expression" and "set a new expression of type-whatever" with this _if_. Notice that it must be after `createUndoCommand()` and before the `expressionCommand.execute()` to keep the undo/redo working.